src/react-native-animated-charts/src/charts/linear/ChartPathProvider.tsx

```diff
@@ -0,0 +1,206 @@
+      return shape.curveLinear;
+    case CurveType.monotone:
+      return shape.curveMonotoneX;
+    case CurveType.natural:
+      return shape.curveNatural;
+    case CurveType.step:
+      return shape.curveStep;
+
+    default:
+      return shape.curveBasis;
+  }
+}
+
+export const ChartPathProvider = React.memo<ChartPathProviderProps>(
+  ({ children, data, width = WIDTH, height = HEIGHT, yRange }) => {
+    const progress = useSharedValue(1);
+    const dotScale = useSharedValue(0);
+    const isActive = useSharedValue(false);
+    const originalX = useSharedValue('');
+    const originalY = useSharedValue('');
+    const pathOpacity = useSharedValue(1);
+    const state = useSharedValue(0);
+    const positionX = useSharedValue(0);
+    const positionY = useSharedValue(0);
+
+    const getScales = useCallback(
+      ({ data, width, height, yRange }: CallbackType): PathScales => {
+        const x = data.points.map(item => item.x);
+        const y = data.points.map(item => item.y);
+
+        const smallestX = Math.min(...x);
+        const smallestY = Math.min(...y);
+        const greatestX = Math.max(...x);
+        const greatestY = Math.max(...y);
+
+        const scaleX = scaleLinear()
+          .domain([smallestX, greatestX])
+          .range([0, width]);
+
+        const scaleY = scaleLinear()
+          .domain(yRange ?? [smallestY, greatestY])
+          .range([height, 0]);
+
+        return {
+          scaleY,
+          scaleX,
+        };
+      },
+      []
+    );
+
+    const createPath = useCallback(
+      ({ data, width, height, yRange }: CallbackType): PathData => {
+        const { scaleY, scaleX } = getScales({
+          data,
+          width,
+          height,
+          yRange,
+        });
+
+        if (!data.points.length) {
+          return {
+            path: '',
+            parsed: null,
+            points: [],
+            data: [],
+          };
+        }
+
+        const points: Point[] = [];
+
+        const { greatestY, smallestY } = findYExtremes(data.points) as {
+          greatestY: Point;
+          smallestY: Point;
+        };
+        const smallestX = data.points[0];
+        const greatestX = data.points[data.points.length - 1];
+
+        for (let i = 0; i < data.points.length; i++) {
+          points.push({
+            x: scaleX(data.points[i].x),
+            y: scaleY(data.points[i].y),
+          });
+        }
+
+        const path = shape
+          .line()
+          .x((item: Point) => scaleX(item.x))
+          .y((item: Point) => scaleY(item.y))
+          .curve(getCurveType(data.curve!))(data.points) as string;
+
+        const parsed = redash.parse(path);
+
+        return {
+          path,
+          parsed,
+          points,
+          data: data.points,
+          smallestX,
+          smallestY,
+          greatestX,
+          greatestY,
+        };
+      },
+      []
+    );
+
+    const initialized = useRef(false);
+
+    const initialPath = useMemo(
```

Comment:
how about useRef with a log to set the initial value only once? I don't this useMemo is really for this

Reply:
According to docs, this might rerender extra time. 
Once, I've seen the pattern:

```
const initialPath = useState(() => { ... 
```